libdnf5/base/goal.cpp

```diff
@@ -625,10 +627,15 @@ GoalProblem Goal::Impl::add_transaction_replay_specs_to_goal(base::Transaction &
     if (!serialized_transaction) {
         return GoalProblem::NO_PROBLEM;
     }
-    libdnf5::GoalJobSettings settings = serialized_transaction->second;
+
+    auto & [replay_path, settings] = *serialized_transaction;
+    utils::fs::File replay_file(replay_path, "r");
+    auto replay_location = replay_path.remove_filename();
+    auto replay = transaction::parse_transaction_replay(replay_file.read());
```

Comment:
It's a bit confusing that `--store` takes a directory path but `history replay` needs a path to the `transaction.json`, which must be in a directory along with the `packages` and `comps`.

Currently, if you pass a directory to `history replay` it throws:

```
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
```

IMO `history replay` should only accept a directory, unless we really want to support bare `transaction.json`s.

Reply:
I agee, but actually I didn't want to include the `history replay` commit :facepalm:, we want to move/rename the command. It was just for testing, I will drop it for now.

Reply:
OK gotcha. Another thing I noticed is that the RPMs in the stored transaction get deleted during replay, which may be undesired. We'll figure it out later.